src/components/entityset/EntitySet.js

```diff
@@ -0,0 +1,144 @@
+// @flow
+
+import React, { PropTypes } from 'react';
+import { Link } from 'react-router';
+import { SplitButton, Button, MenuItem } from 'react-bootstrap';
+
+import { DataApi, EntityDataModelApi, PermissionsApi } from 'loom-data';
+
+import FileConsts from '../../utils/Consts/FileConsts';
+import PageConsts from '../../utils/Consts/PageConsts';
+import { EntitySet } from './EntitySetStorage';
+import styles from './entityset.module.css';
+
+const MAX_DESCRIPTION_LENGTH = 300;
+
+class ExpandableText extends React.Component {
+  static propTypes = {
+    text: PropTypes.string,
+    maxLength: PropTypes.number
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
`ExpandableText` is a self-contained utility component - similar to a tooltip. We could put `isOpen` in the Redux store, but it's not that relevant to the application state, so it's a lot of unnecessary overhead.